src/loki-fs-structured-adapter.js

```diff
@@ -0,0 +1,255 @@
+
+/*
+  Loki (node) fs structured Adapter (need to require this script to instance and use it).
+
+  This adapter will save database container and each collection to separate files and
+  save collection only if it is dirty.  It is also designed to use a destructured serialization
+  method intended to lower the memory overhead of json serialization.
+
+  This adapter utilizes ES6 generator/iterator functionality to stream output and
+  uses node linereader module to stream input.  This should lower memory pressure
+  in addition to individual object serializations rather than loki's default deep object
+  serialization.
+*/
+
+(function (root, factory) {
+    if (typeof define === 'function' && define.amd) {
+        // AMD
+        define([], factory);
+    } else if (typeof exports === 'object') {
+        // Node, CommonJS-like
+        module.exports = factory();
+    } else {
+        // Browser globals (root is window)
+        root.LokiFsStructuredAdapter = factory();
+    }
+}(this, function () {
+  return (function() {
+
+    const fs = require('fs');
+    const readline = require('readline');
+    const stream = require('stream');
+
+    /**
+     * Loki structured (node) filesystem adapter class.
+     *     This class fulfills the loki 'reference' abstract adapter interface which can be applied to other storage methods.
+     *
+     * @constructor LokiFsStructuredAdapter
+     *
+     */
+    function LokiFsStructuredAdapter()
+    {
+        this.mode = "reference";
+        this.dbref = null;
+        this.dirtyPartitions = [];
+    }
+
+    /**
+     * Generator for constructing lines for file streaming output of db container or collection.
+     *
+     * @param {object=} options - output format options for use externally to loki
+     * @param {int=} options.partition - can be used to only output an individual collection or db (-1)
+     *
+     * @returns {string|array} A custom, restructured aggregation of independent serializations.
+     * @memberof LokiFsStructuredAdapter
+     */
+    LokiFsStructuredAdapter.prototype.generateDestructured = function*(options) {
+      var idx, sidx;
+      var dbcopy;
+
+      options = options || {};
+
+      if (!options.hasOwnProperty("partition")) {
+        options.partition = -1;
+      }
+
+      // if partition is -1 we will return database container with no data
+      if (options.partition === -1) {
+        // instantiate lightweight clone and remove its collection data
+        dbcopy = this.dbref.copy();
+
+        for(idx=0; idx < dbcopy.collections.length; idx++) {
+          dbcopy.collections[idx].data = [];
+        }
+
+        yield dbcopy.serialize({
+          serializationMethod: "normal"
+        });
+
+        return;
+      }
+
+      // 'partitioned' along with 'partition' of 0 or greater is a request for single collection serialization
+      if (options.partition >= 0) {
```

Comment:
this condition is redundant, `options.partition` can only be >= -1, if it's not -1 then this block gets executed, am I getting this wrong?